src/eal.cpp

```diff
@@ -80,32 +83,62 @@ namespace eal {
       throw invalid_argument("Invalid input coeffs, expected three vectors.");
     }
 
-    vector<double> coordinate = {0.0, 0.0, 0.0};  
-    coordinate[0] = evaluatePolynomial(coeffs[0], time); // X 
+    vector<double> coordinate = {0.0, 0.0, 0.0};
+    coordinate[0] = evaluatePolynomial(coeffs[0], time); // X
     coordinate[1] = evaluatePolynomial(coeffs[1], time); // Y
     coordinate[2] = evaluatePolynomial(coeffs[2], time); // Z
 
     return coordinate;
   }
 
 
-  // Velocity Function 
+  // Velocity Function
   // Takes the coefficients from the position equation
   vector<double> getVelocity(vector<vector<double>> coeffs, double time) {
     vector<double> coordinate = {0.0, 0.0, 0.0};
     return coordinate;
   }
 
-  // Rotation Data Functions
 
   // Rotation Data Functions
-  vector<double> getRotation(string from, string to, vector<vector<double>> rotations,
+  vector<double> getRotation(vector<vector<double>> rotations,
                              vector<double> times, double time,  interpolation interp) {
-    vector<double> coordinate = {0.0, 0.0, 0.0};
+    // Check that all of the data sizes are okay
+    // TODO is there a cleaner way to do this? We're going to have to do this a lot.
+    if (rotations.size() != 4) {
+     throw invalid_argument("Invalid input rotations, expected four vectors.");
+    }
+
+    // Alot of copying and reassignment becuase conflicting data types
+    // probably should rethink our vector situation to guarentee contiguous
+    // memory. Should be easy to switch to a contiguous column-major format
+    // if we stick with Eigen.
+    double data[] = {0,0,0,0};
+    for (size_t i = 0; i<rotations[0].size(); i++) {
+      Eigen::Quaterniond quat(rotations[0][i], rotations[1][i], rotations[2][i], rotations[3][i]);
+      quat.normalize();
+      rotations[0][i] = quat.w();
+      rotations[1][i] = quat.x();
+      rotations[2][i] = quat.y();
+      rotations[3][i] = quat.z();
```

Comment:
Bad implementation, total GSD edition. I believe the correct way to do this would be to use column-major vectors if we do decide to use Eigen for matrix operations so we can avoid unnecessary allocs and copies.